include/souper/Parser/Parser.h

```diff
@@ -42,6 +42,7 @@ struct ParsedReplacement {
   }
   std::string getLHSString(ReplacementContext &Context,
                            bool printNames = false) const {
+    Mapping.LHS->DemandedBitsVal = llvm::APInt::getAllOnesValue(Mapping.LHS->Width);
```

Comment:
this line looks very out of place, can you explain why this one variable wants to be initialized here when none of these functions initializes anything else?